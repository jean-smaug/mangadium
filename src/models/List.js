import { types, flow, applySnapshot } from 'mobx-state-tree'
import { arrayOf, shape } from 'prop-types'
import { Manga, MangaPropTypes } from './Manga'
import request from '../utils/request'

export const ListPropTypes = shape({ mangas: arrayOf(MangaPropTypes) })

export const List = types
  .model({
    mangas: types.optional(types.array(Manga), [])
  })
  .actions(self => ({
    afterCreate () {
      self.hyrdate()
    },
    hyrdate: flow(function * () {
      try {
        const mangas = yield request.getTopMangas()

        self.setMangas(mangas)
      } catch (error) {
        console.error(error)
      }
    }),
    setMangas (mangas) {
      applySnapshot(self.mangas, mangas)
    }
  }))

export default List
