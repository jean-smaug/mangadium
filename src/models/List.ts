import { types, flow, applySnapshot } from 'mobx-state-tree'
import { arrayOf } from 'prop-types'
import { Manga, MangaInterface } from './Manga'
import request from '../utils/request'

export interface ListInterface {
  mangas: MangaInterface[]
}

export const List = types
  .model({
    mangas: types.optional(types.array(Manga), [])
  })
  .actions(self => ({
    afterCreate () {
      self.hyrdate()
    },
    hyrdate: flow(function * () {
      try {
        const mangas = yield request.getTopMangas()

        self.setMangas(mangas)
      } catch (error) {
        console.error(error)
      }
    }),
    setMangas (mangas) {
      applySnapshot(self.mangas, mangas)
    }
  }))
