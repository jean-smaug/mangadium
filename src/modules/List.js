import React, { Component, Fragment } from "react";
import { observer, inject } from "mobx-react";

import Card from "../components/Card";

@inject("list", "detail")
@observer
class Manga extends Component {
  render() {
    const { list, detail } = this.props;
    return (
      <Fragment>
        {list.current !== null ? (
          <div>Hello</div>
        ) : (
          <div className="columns is-multiline">
            {list.mangas.map(item => (
              <div
                key={item.id}
                style={{
                  height: "200px",
                  overflow: "hidden",
                  textOverflow: "ellipsis"
                }}
                className="column is-half"
                onClick={() => {
                  detail.setManga(item.id);
                }}
              >
                <Card {...item} />
              </div>
            ))}
          </div>
        )}
      </Fragment>
    );
  }
}

export default Manga;
