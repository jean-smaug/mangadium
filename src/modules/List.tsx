import * as React from 'react'
import { observer, inject } from 'mobx-react'

import Card from '../components/Card'
// import { ListPropTypes } from '../models/List'
// import { DetailPropTypes } from '../models/Detail'

@inject('list', 'detail')
@observer
class Manga extends React.Component<{}, {}> {
  render () {
    const { list, detail } = this.props
    return (
      <React.Fragment>
        <div className='container'>
          <div className='columns is-multiline'>
            {list.mangas.map(item => (
              <div key={item.id} className='column is-one-fifth'>
                <Card
                  manga={item}
                  onClick={() => {
                    detail.setManga(item.id)
                  }}
                />
              </div>
            ))}
          </div>
        </div>
      </React.Fragment>
    )
  }
}

export default Manga
