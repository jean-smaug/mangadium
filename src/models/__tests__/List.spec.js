import List from '../List'

describe('<=== List ===>', () => {
  let list = null

  beforeEach(() => {
    list = List.create()
  })

  it('should create store', () => {
    expect(list).toMatchSnapshot()
  })
})
