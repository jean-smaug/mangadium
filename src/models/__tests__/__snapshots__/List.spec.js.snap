// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<=== List ===> should create store 1`] = `
Object {
  "mangas": Array [],
}
`;
